R/calcCapacity.R

```diff
@@ -72,81 +95,19 @@ calcCapacity <- function(subtype) {
-                         REMIND_techs=c("csp", "geohdr", "hydro", "spv", "wind"),
-                         stringsAsFactors = FALSE)
-    # renaming technologies to REMIND naming convention
-    IRENAcap <- rename_dimnames(IRENAcap, dim = 3, query = mapping, from = "IRENA_techs", to="REMIND_techs")
-    IRENAcap <- IRENAcap * 1E-06 # converting MW to TW
-    #overwriting Russia and Japan capacities for wind and spv to avoid REMIND convergence problems
-    # (this is a temporary solution that should be removed once the bounds in REMIND are reworked)
-    # IRENAcap["JPN",2010,"wind"] <- 0.0012
-    # IRENAcap["RUS",2010,"spv"] <- 5e-06
-    # IRENAcap["RUS",2015,"wind"] <- 2e-05
-    # IRENAcap["RUS",2015,"spv"] <- 2e-05
-
-    # Use Openmod capacity values updated by the LIMES team for the European countries.
-    # Year: 2015
-    # Technologies: "tnrs","ngcc","ngt","dot"
-    Openmodcap <- readSource(type="Openmod") # Read Openmod capacities
-    # selecting data used on REMIND # "BAL"
-    Openmodcap <- Openmodcap[c("FIN","NOR","SWE","EST","LVA","LTU","DNK","GBR","IRL","NLD","POL",
-                               "DEU","BEL","LUX","CZE","SVK","AUT","CHE","HUN","ROU","SVN","FRA",
-                               "HRV","BGR","ITA","ESP","PRT","GRC"),,c("tnr","ngcc","ngt","oil")]
-    mapping <- data.frame( Openmod_techs=c("tnr","ngcc","ngt","oil"),
-                           REMIND_techs=c("tnrs","ngcc","ngt","dot"), stringsAsFactors = FALSE)
-    # renaming technologies to REMIND naming convention
-    Openmodcap <- rename_dimnames(Openmodcap, dim = 3, query = mapping, from = "Openmod_techs", to="REMIND_techs")
-    Openmodcap <- Openmodcap * 1E-03 # converting GW to TW
-
-    # Use WEO 2017 data to additional countries: "USA","BRA","RUS","CHN","IND","JPN"
-    # Year: 2015
-    # Technologies: "tnrs","dot"
-    WEOcap <- readSource(type="IEA_WEO",subtype="Capacity") # Read IEA WEO capacities
-    WEOcap <- WEOcap[c("USA","BRA","RUS","CHN","IND","JPN"),2015,c("Nuclear","Oil")] # selecting data used on REMIND
-    mapping <- data.frame( WEO_techs=c("Nuclear","Oil"),
-                           REMIND_techs=c("tnrs","dot"), stringsAsFactors = FALSE)
-    # renaming technologies to REMIND naming convention
-    WEOcap <- rename_dimnames(WEOcap, dim = 3, query = mapping, from = "WEO_techs", to="REMIND_techs")
-    WEOcap <- WEOcap * 1E-03 # converting GW to TW
-
+    
     # merge IRENA, Openmod and WEO capacities data
     output <- new.magpie(cells_and_regions=unique(c(getRegions(IRENAcap),getRegions(Openmodcap),getRegions(WEOcap))),
                          years = unique(c(getYears(IRENAcap),getYears(Openmodcap),getYears(WEOcap))),
-                         names = unique(c(getNames(IRENAcap),getNames(Openmodcap),getNames(WEOcap))),
+                         names = unique(c(getNames(IRENAcap),getNames(Openmodcap),getNames(WEOcap), getNames(CHA.2020.GasData))),
                          fill=0)
```

Comment:
So you define some magpie object, leave it empty, to get it's "names" (one string) from it?

I'd suggest
```
    CHA.2020.GasData <- as.magpie(
      tribble(
        ~region,   ~year,   ~data,      ~value,
        "CHA",     2010,    "gaschp",   0.022,
        "CHA",     2015,    "gaschp",   0.05,
        "CHA",     2020,    "gaschp",   0.054,
        "CHA",     2010,    "ngcc",     0.001,
        "CHA",     2015,    "ngcc",     0.005,
        "CHA",     2020,    "ngcc",     0.01,
        "CHA",     2010,    "ngt",      0.003,
        "CHA",     2015,    "ngt",      0.016,
        "CHA",     2020,    "ngt",      0.026))
…
    output[getRegions(CHA.2020.GasData),getYears(CHA.2020.GasData),
           getNames(CHA.2020.GasData)] <- CHA.2020.GasData
```